internal/app/list/list.go

```diff
@@ -4,45 +4,49 @@ import (
 	"errors"
 	"fmt"
 	"log"
-	"os"
+	"path/filepath"
 
+	"github.com/DanWlker/remind/internal/data"
 	i_error "github.com/DanWlker/remind/internal/error"
-	"github.com/DanWlker/remind/internal/pkg/data"
-	"github.com/DanWlker/remind/internal/pkg/record"
-	"github.com/DanWlker/remind/internal/pkg/shared"
+	"github.com/DanWlker/remind/internal/record"
+	"github.com/DanWlker/remind/internal/shared"
 )
 
 func listOne(pathToFind string) error {
-	projectRecordEntity, errFindProjectRecordEntity := record.GetRecordEntityWithIdentifier(pathToFind)
-	var errRecordDoesNotExist *i_error.RecordDoesNotExistError
-	if errors.As(errFindProjectRecordEntity, &errRecordDoesNotExist) {
-		recordIdentifier := errRecordDoesNotExist.RecordIdentifier
+	projectRecordEntity, err := record.GetRecordEntityWithIdentifier(pathToFind)
+
+	var errRecordDoesNotExist i_error.RecordDoesNotExistError
+	if errors.As(err, &errRecordDoesNotExist) {
+		recordIdentifier := errRecordDoesNotExist.ID
 		if recordIdentifier == "" {
 			recordIdentifier = "$HOME"
 		}
 		fmt.Println("No record linked to this folder found: " + recordIdentifier)
 		return nil
-	} else if errFindProjectRecordEntity != nil {
-		return fmt.Errorf("helper.FindProjectRecordFromFileWith: %w", errFindProjectRecordEntity)
+	}
+	if err != nil {
+		return fmt.Errorf("helper.FindProjectRecordFromFileWith: %w", err)
 	}
 
-	dataFolder, errGetDataFolder := data.GetFolder()
-	if errGetDataFolder != nil {
-		return fmt.Errorf("helper.GetDataFolder: %w", errGetDataFolder)
+	dataFolder, err := data.GetFolder()
+	if err != nil {
+		return fmt.Errorf("helper.GetDataFolder: %w", err)
 	}
 
-	if errPrettyPrintFile := data.PrettyPrintFile(
-		dataFolder+string(os.PathSeparator)+projectRecordEntity.DataFileName,
+	err = data.PrettyPrintFile(
+		filepath.Join(dataFolder, projectRecordEntity.DataFileName),
 		func(todo string, index int) string {
 			return fmt.Sprintf("\t%v. %v", index, todo)
-		}); errPrettyPrintFile != nil {
-		return fmt.Errorf("helper.PrettyPrintDataFile: %w", errPrettyPrintFile)
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("helper.PrettyPrintDataFile: %w", err)
 	}
 	return nil
 }
 
-func listConcurrently(item record.RecordEntity, dataFolder string) (chan string, error) {
-	var c chan string = make(chan string)
+func listConcurrently(item record.RecordEntity, dataFolder string) (<-chan string, error) {
```

Comment:
The caller should expect only to read from the channel returned from this function, not write to it, so return the more-specific type.